File: backend/src/main/java/xyz/engsmyre/moviedescriptiongame/dto/gameId/GameIdConverter.java
package xyz.engsmyre.moviedescriptiongame.dto.gameId;

import java.util.UUID;
import org.springframework.core.convert.converter.Converter;

public class GameIdConverter implements Converter<String, GameId> {
    @Override
    public GameId convert(String source) {
        try {
            UUID gameUUID = UUID.fromString(source);
            return new GameId(gameUUID);
        } catch (IllegalArgumentException e) {
            System.out.println("could not convert from String to GameId, internal error: " + e.getMessage());
            throw new IllegalArgumentException(e);
        }
    }
}

File: backend/src/main/kotlin/xyz/engsmyre/moviedescriptiongame/dto/gameId/GameIdConverter.kt
package xyz.engsmyre.moviedescriptiongame.dto.gameId

import org.springframework.core.convert.converter.Converter
import java.util.*

class GameIdConverter : Converter<String?, GameId> {
    override fun convert(source: String?): GameId {
        return try {
            val gameUUID = UUID.fromString(source)
            GameId(gameUUID)
        } catch (e: IllegalArgumentException) {
            println("could not convert from String to GameId, internal error: ${e.message}")
            throw IllegalArgumentException(e)
        }
    }
}